src/components/infinite-scroll.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+type InfiniteScrollProps = {
+  onBottomHit: () => void;
+  isLoading: boolean;
+  children: React.ReactNode;
+};
+
+function isBottom(ref: React.RefObject<HTMLDivElement>) {
+  if (!ref.current) {
+    return false;
+  }
+  return ref.current.getBoundingClientRect().bottom <= window.innerHeight;
+}
+
+export const InfiniteScroll: React.FC<InfiniteScrollProps> = (props) => {
+  const contentRef = React.useRef(null);
+
+  const handleLoadMore = () => {
+    if (!props.isLoading && isBottom(contentRef)) {
+      props.onBottomHit();
+    }
+  };
+
+  React.useEffect(() => {
+    handleLoadMore();
+    document.addEventListener('scroll', handleLoadMore);
+    return () => document.removeEventListener('scroll', handleLoadMore);
+  }, [props]);
```

Comment:
geralmente a gente separa os useEffects, pra ficar claro o que cada um faz, sabe? então aqui ficaria algo assim
```suggestion
// esse cuida do fetch 'inicial' - quando a página aparece a primeira vez
React.useEffect(() => {
  handleLoadMore();
}, []);

// esse cuida dos eventos de scroll
React.useEffect(() => {
  document.addEventListener('scroll', handleLoadMore);
  return () => document.removeEventListener('scroll', handleLoadMore);
}, []);
```